opentelemetry-api/src/opentelemetry/context/propagation/httptextformat.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import typing
+from opentelemetry.trace import SpanContext
+
+Setter = typing.Callable[[object, str, str], None]
+Getter = typing.Callable[[object, str], str]
+
+
+class HTTPTextFormat(abc.ABC):
+    """API for propagation of spans via headers.
+
+    This class provides an interface that enables extracting and injecting
+    trace state into headers of HTTP requests. Http frameworks and client
+    can integrate with HTTPTextFormat by providing the object containing the
+    headers, and a getter and setter function for the extraction and
+    injection of values, respectively.
+
+    Example::
+
+        import flask
+        import requests
+        from opentelemetry.context.propagation import HTTPTextFormat
+
+        PROPAGATOR = HTTPTextFormat()
+
+
+
+        def get_header_from_flask_request(request, key):
+            return request.headers[key]
+
+        def set_header_into_requests_request(request: requests.Request,
+                                             key: str, value: str):
+            request.headers[key] = value
+
+        def example_route():
+            span_context = PROPAGATOR.extract(
+                get_header_from_flask_request,
+                flask.request
+            )
+            request_to_downstream = requests.Request(
+                "GET", "http://httpbin.org/get"
+            )
+            PROPAGATOR.inject(
+                span_context,
+                set_header_into_requests_request,
+                request_to_downstream
+            )
+            session = requests.Session()
+            session.send(request_to_downstream.prepare())
+
+
+    .. _Propagation API Specification:
+       https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-propagators.md
+
+    Enabling this flexi
+    """
+    @abc.abstractmethod
+    def extract(self, get_from_carrier: Getter,
+                carrier: object) -> SpanContext:
```

Comment:
I think this is intentional, but may be worth noting that this class is effectively java's `HttpTraceContext`, which implements `HttpTextFormat<SpanContext>`. We're unlikely to match java's class structure here, but it is confusing that this class uses the same name. Also confusing that it lives in the context package but only extracts trace data from `SpanContext`, which belongs to the tracing package.

Reply:
That is a great point. It let me to some thinking and some issues in the OT specification that I've filed.

By the sound of it's worth an exploration in propagators a bit to figure out the right behavior. For now I'll leave it as is and pick up conversations in the tickets and https://github.com/open-telemetry/opentelemetry-specification/issues/211